### PR TITLE
fix: Event program relationships can now be added through maintenance app

### PR DESCRIPTION
**Pull request by stian-sandvold**
In the case of a program without registration, there is only one program stage. As such, the UI only presents the program, and sends it in.

This fix will populate the ProgramStage value with that single program stage of an event program if its not present.

### CODE REVIEW

**Review by enricocolasante — APPROVED** [review]

**Review by lucaCambi77 — COMMENTED** [review]
Any way to add a test?

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/event/RelationshipTypeController.java [hunk 1]
```diff
@@ -41,4 +44,53 @@
...
+     * We do this because programs without registration only have a single
+     * program stage
+     *
+     * @param constraint
+     */
+    private void handleEventProgramRelationshipConstraint( RelationshipConstraint constraint )
+    {
+        if ( constraint.getProgram() != null && constraint.getProgramStage() == null )
+        {
+            Program program = programService.getProgram( constraint.getProgram().getUid() );
```
Comment (teleivo) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/event/RelationshipTypeController.java:88]: Can we safely assume the idScheme is UID here?

File: dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/event/RelationshipTypeController.java [hunk 2]
```diff
@@ -41,4 +44,53 @@
...
+
+            if ( program.isWithoutRegistration() )
+            {
+                constraint.setProgramStage( program.getProgramStageByStage( 1 ) );
```
Comment (teleivo) [inline dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/event/RelationshipTypeController.java:92]: What does the `1` mean here? Might be worth documenting in a constant.

**Review by teleivo — COMMENTED** [review]
This change only affects old tracker, is that intentional? 😄 
👍🏻 to @lucaCambi77 for adding a test 😋

**Pull request closed without merging**